publisher.go

```diff
@@ -0,0 +1,76 @@
+package google_pubsub
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/go-playground/validator/v10"
+	"github.com/google/uuid"
+)
+
+type (
+	PublisherConfig struct {
+		ProjectID string
+	}
+
+	Publisher struct {
+		config    PublisherConfig
+		provider  *pubsub.Client
+		validator *validator.Validate
+	}
+
+	Content struct {
+		data []byte
+	}
+
+	Header  map[string]string
+	Message struct {
+		// Enable to identify message, if order must be respect this field it's used then.
+		OriginalKey string
+		Topic       string  `validate:"required"`
+		Content     Content `validate:"required"`
+		Headers     Header
+	}
+)
+
+func NewContent() Content { return Content{} }
+
+func (c Content) Marshal(content any) Content {
```

Comment:
```suggestion
func (c Content) Marshal() Content {
```

and return an error if `c.data` is empty.
Also, return the error if `json.Marshal` fails